### PR TITLE
build: cmake: sync with `configure.py` (3/n)

### PR DESCRIPTION
**Pull request by tchaikov**
* build: cmake: add test
* build: cmake: expose the bridged rust library
* build: cmake: correct library path
* build: cmake: add missing source files
* build: cmake: put generated sources into ${scylla_gen_build_dir}
* build: cmake: silence -Wuninitialized warning
* build: cmake: extract idl library out
* build: cmake: ignore -Wparentheses-equality

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -98,14 +98,12 @@ set(scylla_sources
     auth/service.cc
     auth/standard_role_manager.cc
     auth/transitional.cc
-    bytes.cc
```
Comment (denesb) [inline CMakeLists.txt:156]: Why are these files added to the utils lib?
If you think they belong there conceptionally, please move them to the `utils/` directory first (in a separate PR please).

File: CMakeLists.txt [hunk 2]
```diff
@@ -340,49 +337,9 @@ set(scylla_sources
     transport/event_notifier.cc
     transport/messages/result_message.cc
     transport/server.cc
-    types.cc
```
Comment (denesb) [inline CMakeLists.txt:380]: This is not some random utility, this is the implementation of the CQL type system. This should be moved to `types/`.

**Review by denesb — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (tchaikov) [reply CMakeLists.txt:156]: i was trying to fix the dependencies of the tests. as in the C++20 module era, we are likely have a bunch of modules, and tests are going to import (some of) them for exercising the exposed functionalities. and the tests are not likely to need to link against the whole scylla blob in a whole sale for testing a tiny piece of functionality.

but i failed, this dependencies between the translation units in `scylla_sources` and the ones in the subdirectories is not a tree. sometimes, they form circular graph.

now that the change does not work at all. i will drop this change. and keep it in `scylla_sources` in this pull request, and revisit this problem later.

File: CMakeLists.txt (hunk 2 above)
Comment (tchaikov) [reply CMakeLists.txt:380]: i will defer the change to a follow-up commit. we do need to fix it.

File: CMakeLists.txt (hunk 1 above)
Comment (denesb) [reply CMakeLists.txt:156]: > sometimes, they form circular graph

Yes, this is a known problem that needs to be sorted out before we use modules, which requires a DAG dependencies.

**Comment by tchaikov** [comment]
changelog:

- put some of the sources previously in `scylla_sources` back. we should fix this by moving them to the right subdirectory first.

@denesb hi Botond, could you take another look?

**Review by denesb — APPROVED** [review]

**Comment by denesb** [comment]
@tchaikov please rebase, there are conflicts. Possibly with next, so you might have to wait until promotion completes.

**Comment by tchaikov** [comment]
@denesb Botond, rebased against master HEAD and repushed.

**Comment by denesb** [comment]
Queued. Not waiting on CI as it is not checking CMakeLists.txt anyway.

File: CMakeLists.txt (hunk 2 above)
Comment (tchaikov) [reply CMakeLists.txt:379]: see #12926
